parser/src/main/java/com/microsoft/gctoolkit/parser/UnifiedGenerationalParser.java

```diff
@@ -481,12 +483,21 @@ private FullGC fillOutFullGC(FullGC event, GenerationalForwardReference values)
     }
 
     private FullGC buildFullGC(GenerationalForwardReference forwardReference) {
+        FullGC gc;
         switch (forwardReference.getGarbageCollectionType()) {
             case PSFull:
-                return fillOutFullGC(new PSFullGC(forwardReference.getStartTime(), forwardReference.getGCCause(), forwardReference.getDuration()), forwardReference);
+                if ( forwardReference.getGCCause().equals(GCCause.JAVA_LANG_SYSTEM))
```

Comment:
It feels like some of this is DRY / could be squashed down, on the other hand it's readable, so that wins!